.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,28 @@
+Thank you for contributing to JanusGraph.
+
+In order to streamline the review of the contribution we ask you
+to ensure the following steps have been taken:
+
+### For all changes:
+- [ ] Is there an issue associated with this PR? Is it referenced 
+     in the commit message?
+
+- [ ] Does your PR title ends with #xyz where xyz is the issue number you are trying to resolve?
+
+- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?
```

Comment:
The `master` branch in JanusGraph is already protected against forced pushes, and @kottmann makes a good point that this is orthogonal — we need to make sure it will be a clean merge.